internal/webserver/webserver.go

```diff
@@ -217,40 +228,45 @@ func (n kubeFilter) namespacesHandler(writer http.ResponseWriter, request *http.
 	n.handleRequest(request, username, selector)
 }
 
-func (n kubeFilter) isNamespaceListing(fn http.HandlerFunc) http.HandlerFunc {
-	return func(writer http.ResponseWriter, request *http.Request) {
-		if req.NewHttp(request, n.usernameClaimField).IsNamespaceListing() {
-			log.V(3).Info("Handling /api/v1/namespaces")
-			fn(writer, request)
-		}
-	}
-}
-
-func (n kubeFilter) isNodeListing(fn http.HandlerFunc) http.HandlerFunc {
-	return func(writer http.ResponseWriter, request *http.Request) {
-		if req.NewHttp(request, n.usernameClaimField).IsNodeListing() {
-			log.V(3).Info("Handling /api/v1/nodes")
-			fn(writer, request)
-		}
-	}
-}
-
 func (n kubeFilter) Start(ctx context.Context) error {
 	r := mux.NewRouter()
-	r.HandleFunc("/api/v1/namespaces", func(writer http.ResponseWriter, request *http.Request) {
-		n.isNamespaceListing(n.checkJwt(n.namespacesHandler))(writer, request)
-		n.reverseProxyFunc(writer, request)
-	})
-	r.HandleFunc("/api/v1/nodes", func(writer http.ResponseWriter, request *http.Request) {
-		n.isNodeListing(n.checkJwt(n.nodesHandler))(writer, request)
-		n.reverseProxyFunc(writer, request)
-	})
+	r.StrictSlash(true)
+	r.Use(n.reverseProxyMiddleware)
```

Comment:
I love this middleware, because in the end, we're telling each route to perform the reverse-proxy `ServeHTTP` method, delegating to the handlers just the request manipulation: less code is better.